libs/designsystem/modal/src/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -404,17 +404,29 @@ export class ModalWrapperComponent
   private observeTitleContentChanges() {
     this.mutationObserverUnobserveFn = observeContent(
       this.ionTitleElement.nativeElement,
-      this.setTitleAsAriaLabel
+      this.setAriaLabelFromTitleContent
     );
   }
 
-  private setTitleAsAriaLabel = () => {
+  private setAriaLabelFromTitleContent = () => {
     const titleTextContent = this.ionTitleElement?.nativeElement.textContent;
     if (this.ionModalDialog && titleTextContent) {
       this.renderer.setAttribute(this.ionModalDialog, 'aria-label', titleTextContent);
     }
   };
 
+  private setAriaLabel() {
+    const ariaLabel = this.config.htmlAttributes?.['aria-label'];
+    const ionModalElementDialog = getIonModalDialogAncestor(this.ionModalElement);
+
+    if (ionModalElementDialog && ariaLabel) {
+      this.renderer.setAttribute(ionModalElementDialog, 'aria-label', ariaLabel);
+    } else {
+      this.observeTitleContentChanges();
+      this.setAriaLabelFromTitleContent();
```

Comment:
Really nitty gritty 🙈 😅, but somehow I think this would read better in reverse order, i.e. "set it, then observe"   
```suggestion
      this.setAriaLabelFromTitleContent();
      this.observeTitleContentChanges();
```